modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnPermissionValidationRuleEngineImpl.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.internal.object.validation.rule;
+
+import com.liferay.commerce.constants.CommerceActionKeys;
+import com.liferay.commerce.constants.CommerceConstants;
+import com.liferay.commerce.constants.CommerceReturnConstants;
+import com.liferay.object.scope.ObjectDefinitionScoped;
+import com.liferay.object.validation.rule.ObjectValidationRuleEngine;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Crescenzo Rega
+ */
+@Component(service = ObjectValidationRuleEngine.class)
+public class CommerceReturnPermissionValidationRuleEngineImpl
+	implements ObjectDefinitionScoped, ObjectValidationRuleEngine {
```

Comment:
I think we can just make this extend BaseObjectValidationRuleEngineImpl so we can follow the pattern that bchan set

Reply:
Validators that extend the `BaseObjectValidationRuleEngineImpl` are validators that validate a specific field, these new validators instead do not validate a field but prevent the modification of some fields if the Return status is different from DRAFT and if the user does not have the MANAGE_RETURNS permission